packages/connectkit/src/hooks/useExternalPaymentOptions.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  ExternalPaymentOptionMetadata,
+  ExternalPaymentOptions,
+  PlatformType,
+} from "@daimo/common";
+import { useEffect, useState } from "react";
+import { trpc } from "../utils/trpc";
+
+const DEFAULT_EXTERNAL_PAYMENT_OPTIONS = [
+  ExternalPaymentOptions.Coinbase,
+  ExternalPaymentOptions.Binance,
+  ExternalPaymentOptions.Daimo,
+  ExternalPaymentOptions.RampNetwork,
+];
+
+export function useExternalPaymentOptions({
+  filterIds,
+  usdRequired,
+  platform,
+}: {
+  filterIds: string[] | undefined;
+  usdRequired: number | undefined;
+  platform: PlatformType | undefined;
+}) {
+  const [options, setOptions] = useState<ExternalPaymentOptionMetadata[]>([]);
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    const refreshExternalPaymentOptions = async (usd: number) => {
+      if (!platform) return;
+
+      setLoading(true);
+      const options = await trpc.getExternalPaymentOptions.query({
```

Comment:
not for this PR, but we could turn on `@trpc/react` to get hooks with loading + error state out of the box

Reply:
when using a separate `setLoading(...)`, use
```
try {
   await ...
   setOptions(..)
} finally {
  setLoading(false);
}
```
 
 to avoid infinite spinner on network/api error

Reply:
done